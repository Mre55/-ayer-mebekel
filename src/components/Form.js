import React, { useState } from 'react';
import { PropTypes } from 'prop-types';

const Form = (props) => {
  const { seeMore, fetchNewData } = props;

  const [lat, setLat] = useState();
  const [long, setLong] = useState();

  const handleLatChange = (e) => {
    setLat(e.target.value);
  };
  const handleLongChange = (e) => {
    setLong(e.target.value);
  };

  return (
    <div>
      <h2 className="text-4xl font-bold">Forecast air pollution data</h2>
      <form
        onSubmit={async (e) => {
          e.preventDefault();
          await fetchNewData(e.target[0].value, e.target[1].value);
        }}
      >
        <label htmlFor="latitude">
          Latitude
          <br />
          <input
            className="border-solid border-2 border-sky-500"
            name="lat"
            id="latitude"
            type="number"
            value={lat}
            placeholder="Enter a Latitude..."
            onChange={() => handleLatChange}
            required
          />
          <br />
        </label>
        <label htmlFor="longitude">
          Longitude
          <br />
          <input
            className="border-solid border-2 border-sky-500"
            name="long"
            id="longitude"
            type="number"
            placeholder="Enter a Longitude..."
            value={long}
            onChange={() => handleLongChange}
            required
          />
          <br />
        </label>
        <input
          onClick={() => seeMore('New')}
          className="bg-blue-600 text-white p-2 w-40 text-lg  rounded mt-4"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  );
};

Form.propTypes = {
  seeMore: PropTypes.func.isRequired,
  fetchNewData: PropTypes.func.isRequired,
};

export default Form;
