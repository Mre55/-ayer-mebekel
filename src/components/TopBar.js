import React from 'react';

const TopBar = () => (
  <nav className="">
    <div className="flex items-end justify-center py-2 ring-4 bg-[#4369b2]">
      <h2 className="text-2xl font-bold text-white">የአየር ብክለት </h2>
      <h2 className="text-lg font-bold text-white"> (Air pollution)</h2>
    </div>
  </nav>
);
export default TopBar;
