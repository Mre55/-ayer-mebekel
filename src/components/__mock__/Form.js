import React from 'react';

const Form = () => (
  <div className="flex bg-[#5788e6] w-screen text-white px-3 items-center md: justify-evenly">
    <h2 className="text-3xl font-bold">Forecast air pollution data</h2>
    <div className="text-center py-4">
      <form
        onSubmit={async (e) => {
          e.preventDefault();
        }}
        noValidate
      >
        <label htmlFor="latitude">
          <p className="text-lg">Latitude</p>
          <input
            className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
            name="lat"
            id="latitude"
            type="number"
            value="lat"
            placeholder="Enter a Latitude..."
            required
          />
          <br />
        </label>
        <label htmlFor="longitude">
          <p className="text-lg">Longitude</p>
          <input
            className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
            name="long"
            id="longitude"
            type="number"
            placeholder="Enter a Longitude..."
            value="long"
            required
          />
          <br />
        </label>
        <input
          className="bg-blue-600 text-white w-20 text-lg rounded mt-4"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>
);

export default Form;
