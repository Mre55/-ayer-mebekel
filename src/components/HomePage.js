import React, { useEffect, useState } from 'react';

import { useDispatch, useSelector } from 'react-redux';
import { fetchWeatherWithInput, fetchWeather } from '../redux/weather/weatherReducer';

import DetailsPage from './DetailsPage';
import Form from './Form';
import CurrentPollutionData from './CurrentPollutionData';
import store from '../redux/configureStore';
import backIcon from '../images/backIcon.png';

const HomePage = () => {
  const dispatch = useDispatch();

  const [detailPage, setDetailPage] = useState(false);

  useEffect(() => {
    if (Object.keys(store.getState().weatherReducer).length === 0) {
      dispatch(fetchWeather());
    }
  }, []);

  const seeMore = (city) => {
    if (city === 'Ethiopia') {
      dispatch(fetchWeatherWithInput(42, 53));
    } else if (city === 'Brazil') {
      dispatch(fetchWeatherWithInput(8, 37));
    } else if (city === 'USA') {
      dispatch(fetchWeatherWithInput(22, 44));
    } else if (city === 'Canada') {
      dispatch(fetchWeatherWithInput(7, 39));
    }
    setDetailPage(!detailPage);
  };

  const weatherReducer = useSelector((state) => state.weatherReducer);

  const { coord = {}, list = [] } = weatherReducer;

  return (
    <div>
      <div className={`${detailPage ? 'hidden' : ''}`}>
        <Form seeMore={seeMore} />

        <div className={`${detailPage ? 'hidden' : ''}`}>
          <CurrentPollutionData
            seeMore={seeMore}
          />
        </div>
      </div>
      <div className={`${detailPage ? '' : 'hidden'}`}>
        <div className="flex items-center gap-14 bg-[#35538c] text-white px-3 ">
          <button className="w-10" onClick={() => seeMore()} type="button">
            <img src={backIcon} alt="back icon" />
          </button>
          <p className="text-sm text-white py-2 w-screen">
            DATA ABOUT POLLUTING GASES
          </p>
        </div>
        {weatherReducer?.coord && (<DetailsPage coord={coord} list={list} />)}
      </div>
    </div>
  );
};

export default HomePage;
