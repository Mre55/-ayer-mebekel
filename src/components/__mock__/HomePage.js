import React from 'react';

import { useDispatch } from 'react-redux';
import { fetchWeatherWithInput } from '../../redux/weather/weatherReducer';

import Form from '../Form';
import backIcon from '../../images/backIcon.png';

const HomePage = () => {
  const dispatch = useDispatch();

  const fetchDataByInput = async (lat, long) => {
    dispatch(fetchWeatherWithInput(lat, long));
  };

  const seeMore = (city) => {
    if (city === 'Addis Ababa') {
      dispatch(fetchWeatherWithInput(4, 5));
    } else if (city === 'Hawassa') {
      dispatch(fetchWeatherWithInput(8, 37));
    } else if (city === 'Hossana') {
      dispatch(fetchWeatherWithInput(22, 44));
    } else if (city === 'London') {
      dispatch(fetchWeatherWithInput(17, 9));
    }
  };

  return (
    <div>
      <div className={'\'detailPage\' ? \'hidden\' : \'\''}>
        <Form seeMore={seeMore} fetchDataByInput={fetchDataByInput} />

        <div className={'\'detailPage\' ? \'hidden\' : \'\''} />
      </div>
      <div className={'\'detailPage\' ? \'\' : \'hidden\''}>
        <div className="flex items-center gap-14 bg-[#35538c] text-white px-3 ">
          <button className="w-10" onClick={() => seeMore()} type="button">
            <img src={backIcon} alt="back icon" />
          </button>
          <p className="text-sm text-white py-2 w-screen">
            DATA ABOUT POLLUTING GASES
          </p>
        </div>
      </div>
    </div>
  );
};

export default HomePage;
