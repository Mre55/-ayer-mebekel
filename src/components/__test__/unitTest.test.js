
import Form from "../__mock__/Form";
import TopBar from "../TopBar";
import HomePage from "../__mock__/HomePage";

import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';

describe("Home page component snapshot tests", () => {
  const mockStore = configureStore();

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <HomePage />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <Form />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <TopBar />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
