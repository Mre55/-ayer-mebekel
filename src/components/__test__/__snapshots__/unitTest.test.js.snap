// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component snapshot tests renders correctly 1`] = `
<div>
  <div
    className="'detailPage' ? 'hidden' : ''"
  >
    <div
      className="flex bg-[#5788e6] w-screen text-white px-3 items-center md:justify-evenly py-6"
    >
      <h2
        className="text-3xl font-bold"
      >
        Forecast air pollution data
      </h2>
      <div
        className="text-center py-4"
      >
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="latitude"
          >
            <p
              className="text-lg"
            >
              Latitude
            </p>
            <input
              className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
              id="latitude"
              name="lat"
              onChange={[Function]}
              placeholder="Enter a Latitude..."
              required={true}
              type="number"
            />
            <br />
          </label>
          <label
            htmlFor="longitude"
          >
            <p
              className="text-lg"
            >
              Longitude
            </p>
            <input
              className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
              id="longitude"
              name="long"
              onChange={[Function]}
              placeholder="Enter a Longitude..."
              required={true}
              type="number"
            />
            <br />
          </label>
          <input
            className="bg-blue-600 text-white w-20 text-lg rounded mt-4"
            onClick={[Function]}
            type="submit"
            value="Search"
          />
        </form>
      </div>
    </div>
    <div
      className="'detailPage' ? 'hidden' : ''"
    />
  </div>
  <div
    className="'detailPage' ? '' : 'hidden'"
  >
    <div
      className="flex items-center gap-14 bg-[#35538c] text-white px-3 "
    >
      <button
        className="w-10"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="back icon"
          src="backIcon.png"
        />
      </button>
      <p
        className="text-sm text-white py-2 w-screen"
      >
        DATA ABOUT POLLUTING GASES
      </p>
    </div>
  </div>
</div>
`;

exports[`Home page component snapshot tests renders correctly 2`] = `
<div
  className="flex bg-[#5788e6] w-screen text-white px-3 items-center md: justify-evenly"
>
  <h2
    className="text-3xl font-bold"
  >
    Forecast air pollution data
  </h2>
  <div
    className="text-center py-4"
  >
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <label
        htmlFor="latitude"
      >
        <p
          className="text-lg"
        >
          Latitude
        </p>
        <input
          className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
          id="latitude"
          name="lat"
          placeholder="Enter a Latitude..."
          required={true}
          type="number"
          value="lat"
        />
        <br />
      </label>
      <label
        htmlFor="longitude"
      >
        <p
          className="text-lg"
        >
          Longitude
        </p>
        <input
          className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
          id="longitude"
          name="long"
          placeholder="Enter a Longitude..."
          required={true}
          type="number"
          value="long"
        />
        <br />
      </label>
      <input
        className="bg-blue-600 text-white w-20 text-lg rounded mt-4"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`Home page component snapshot tests renders correctly 3`] = `
<nav
  className=""
>
  <div
    className="flex items-end justify-center py-2 ring-4 bg-[#4369b2]"
  >
    <h2
      className="text-2xl font-bold text-white"
    >
      የአየር ብክለት 
    </h2>
    <h2
      className="text-lg font-bold text-white"
    >
       (Air pollution)
    </h2>
  </div>
</nav>
`;
